R/spliceHelperFunctions.R

```diff
@@ -479,43 +487,41 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
   return(relist(unlistedExons, partitioning))
 }
 
-#' Function that selects the first exon from an IRangesList object (exon_rank is required)
-#' @param range IRangesList
+#' Function that selects the first exon from an IRangesList object
+#' @param range IRangesList with elements required to be ordered by coordinates
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
   largestExons <- as.numeric(cumsum(elementNROWS(range)))
```

Comment:
maybe lastExons instead of largestExons? (last exons don't have to be the largest)

Reply:
"largestExons" is renamed to "lastExons".